test/run_tests.bash

```diff
@@ -11,6 +11,12 @@
 SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 TEST_RESULTS_DIR="${SCRIPT_DIR}/results"
 
+# Ensure we run everything for coverage purposes, but ensure failures are returned by the script
+SUCCESS=0
+
+# Do not swallow errors with tee
+set -o pipefail
```

Comment:
Since you're piping to `tee` below I think you need this: https://unix.stackexchange.com/q/14270

So that:
https://github.com/sea-bass/pyrobosim/pull/159/checks#step:7:130